Linting breaks mypy

### Check existing issues

- [x] I've verified that this bug isn't described by any existing issues.

### Bug description

Running the lint command removes mypy as a package.

### Steps to reproduce

```bash
$ make clean && rm -rf .venv
$ make setup lint
$ make lint
```

### Environment

Any build environment.

### snapcraft.yaml

```yaml

```

### Log output

```shell
uv run mypy --show-traceback --show-error-codes snapcraft
error: Failed to spawn: `mypy`
  Caused by: No such file or directory (os error 2)
make: *** [common.mk:150: lint-mypy] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ PROJECT=snapcraft
 COVERAGE_SOURCE="snapcraft,snapcraft_legacy"
 UV_TEST_GROUPS := "--group=dev"
 UV_DOCS_GROUPS := "--group=docs"
-UV_LINT_GROUPS := "--group=lint" "--group=types"
+UV_LINT_GROUPS := "--group=lint" "--group=types" "--group=docs"
 UV_TICS_GROUPS := "--group=tics"
 
 ifneq ($(wildcard /etc/os-release),)

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ PROJECT=snapcraft
 COVERAGE_SOURCE="snapcraft,snapcraft_legacy"
 UV_TEST_GROUPS := "--group=dev"
 UV_DOCS_GROUPS := "--group=docs"
-UV_LINT_GROUPS := "--group=lint" "--group=types" "--group=docs"
+UV_LINT_GROUPS := "--group=lint" "--group=types" $(UV_DOCS_GROUPS)
 UV_TICS_GROUPS := "--group=tics"
 
 ifneq ($(wildcard /etc/os-release),)

--- a/common.mk
+++ b/common.mk
@@ -261,7 +261,7 @@ docs-auto: docs-install  ##- Render the documentation in a live session
 # Override for `install` target in docs project. We still need the Vale setup, so we
 # run that after the parent docs setup.
 .PHONY: docs-install
-docs-install: setup-docs  ##- Set up documentation packages
+docs-install: _setup-docs  ##- Set up documentation packages
 	$(MAKE) -C docs vale-install --no-print-directory
 
 # Alias for `setup-docs`

